repair/row_level.cc

```diff
@@ -2770,6 +2795,29 @@ class row_level_repair {
         master.stats().round_nr_slow_path++;
     }
 
+private:
+    // Update system.repair table
+    future<> update_system_repair_table() {
+        if (_ri.reason != streaming::stream_reason::repair) {
+            return make_ready_future<>();
+        }
+        repair_service& rs = _ri.rs;
+        return rs.update_history(_ri.id.uuid, _table_id, _range, _start_time).then([this] (std::optional<gc_clock::time_point> repair_time_opt) {
+            if (!repair_time_opt) {
+                return make_ready_future<>();
+            }
+            auto repair_time = repair_time_opt.value();
+            repair_update_system_table_request req{_ri.id.uuid, _table_id, _ri.keyspace, _cf_name, _range, repair_time};
+            auto all_nodes = _all_live_peer_nodes;
+            all_nodes.push_back(utils::fb_utilities::get_broadcast_address());
+            return parallel_for_each(all_nodes, [this, req] (gms::inet_address node) {
+                return _ri.messaging.local().send_repair_update_system_table(netw::messaging_service::msg_addr(node), req).then([this, node] (repair_update_system_table_response resp) {
+                    rlogger.debug("repair[{}]: Got repair_update_system_table_response from node {}", _ri.id.uuid, node);
+                });
+            });
```

Comment:
Why not coroutines? less chance of use after free.

Reply:
Do you mean use coroutines inside parallel_for_each? It is a single line call.

Reply:
IIRC, you suggested not to use co_return if it is a single call. Btw, if we use coroutines in the upper level (the entire update_system_repair_table function), do we have alternatives for parallel_for_each?

Reply:
Both the upper level and within the parallel_for_each. It's less boilerplate, yes it's one line, but code to capture "this" and "node".  It's also nicer if we have to extend it later, continuations quickly get tangled.

The recommendation against co_return is for performance sensitive stuff. A coroutine unconditionally allocates a frame. Continuations can allocate nothing (but also have many allocations if there is a long .then chain and the first call blocks).

Reply:
ok, done.